.github/workflows/publish.yaml

```diff
@@ -0,0 +1,76 @@
+name: Publish Docker image
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    paths:
+      - apps/**
+  pull_request:
```

Comment:
Is this going to publish it even if it's on a PR?

Reply:
We should probably just build and not publish on PR

Reply:
When it runs inside a PR it does publish it but the tag looks like this: `1.x.x-prY`
This can be useful for testing

Reply:
I see and do we end up removing these tags?
We could end up with quite a number of these redundant tags..

Reply:
Nope,
We can set it up.
a) Delete the `-prY` tag after the PR is merged.
b) Have a scheduled workflow that prunes last X `-prY` tags

Reply:
Just (a) sounds excellent to me ✅ Let's do that please

Reply:
Looks like there is no API to delete tags yet (https://github.com/docker/hub-feedback/issues/496)
There is a workaround mentioned in this issue that uses Username and Password (not token).

But we have one more issue that haven't thought about, Once this is merged we won't have access to `github.event.number` to know what the tag is named.

Lets rename the "PR" tag to `nightly` (without version prefix) and we overwrite this tag on PRs.
So it will only be a single "redundant" tag?

Reply:
Yeah sounds better to me..i don't suppose we would be having multiple PRs targeting the same app.
So we can expand this if we get to that use case but i am skeptical we will

Reply:
If we have such a use case, we can revisit I guess.

Tag renamed